pkg/reconciler/workload/resource/resource_reconcile.go

```diff
@@ -76,6 +76,15 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
 		return nil
 	}
 
+	// TODO(ncdc): wildcard partial metadata requests are now including CRs that come via APIBindings. Multiple e2e tests
+	// manually assign a scheduling label to a resource but not its namespace. The resources were getting synced based
+	// on the manually assigned label, and staying with that assignment. But now because wildcard partial metadata
+	// lists include these bound CRs, not having the namespace assigned is a problem. For now, don't unassign a
+	// resource that is scheduled if the namespace's assignment is unset.
+	if previousCluster != "" && newCluster == "" {
+		return nil
+	}
+
```

Comment:
this seems wrong. The namespace is driving resource scheduling. We should fix the tests.

Reply:
Which tests do you have in mind? I don't see any setting the label, at least in #1084.

Reply:
It's existing tests that are doing it, such as https://github.com/kcp-dev/kcp/blob/19a74feffe74926ac2e5935ef352260f59d1265f/test/e2e/virtual/syncer/virtualworkspace_test.go#L417. Before the fixes in this PR, the resource controller would never see bound CRs, so it would never try to schedule them. Now it does, and if the namespace has `""`, the resource controller will update the resource to match the namespace's label value. So you either have to manually set the label on the namespace first, or do this.